post-processor/vsphere/post-processor.go

```diff
@@ -190,26 +200,36 @@ func (p *PostProcessor) PostProcess(ctx context.Context, ui packersdk.Ui, artifa
 
 	log.Printf("Starting ovftool with parameters: %s", strings.Join(args, " "))
 
-	ui.Message("Validating Username and Password with dry-run")
+	ui.Message("Validating username and password with dry-run")
 	err = p.ValidateOvfTool(args, ovftool)
 	if err != nil {
 		return nil, false, false, err
 	}
 
-	// Validation has passed, so run for real.
-	ui.Message("Calling OVFtool to upload vm")
-	commandAndArgs := []string{ovftool}
-	commandAndArgs = append(commandAndArgs, args...)
-	comm := &shelllocal.Communicator{
-		ExecuteCommand: commandAndArgs,
-	}
-	flattenedCmd := strings.Join(commandAndArgs, " ")
-	cmd := &packersdk.RemoteCmd{Command: flattenedCmd}
-	log.Printf("[INFO] (vsphere): starting ovftool command: %s", flattenedCmd)
-	err = cmd.RunWithUi(ctx, comm, ui)
-	if err != nil || cmd.ExitStatus() != 0 {
-		return nil, false, false, fmt.Errorf(
-			"Error uploading virtual machine: Please see output above for more information.")
+	// Run ovftool
+	maxRetries := p.config.MaxRetries
+	retryCount := 0
+
+	for retryCount < maxRetries {
```

Comment:
The Packer SDK has a package for implementing retries with a backoff time to prevent immediate retry. Might be work looking into it [here](https://pkg.go.dev/github.com/hashicorp/packer-plugin-sdk@v0.5.2/retry#Config.Run)

```go
	ui.Message("Uploading virtual machine using OVFtool...")
	commandAndArgs := append([]string{ovftool}, args...)
	comm := &shelllocal.Communicator{
		ExecuteCommand: commandAndArgs,
	}
	err = retry.Config{
		Tries: maxRetries,
		ShouldRetry: func(err error) bool {
			return err != nil
		},
		RetryDelay: (&retry.Backoff{InitialBackoff: 200 * time.Millisecond, MaxBackoff: 30 * time.Second, Multiplier: 2}).Linear,
	}.Run(ctx, func(ctx context.Context) error {
		cmd := &packersdk.RemoteCmd{Command: flattenedCmd}
		log.Printf("Starting ovftool command: %s", flattenedCmd)
		err = cmd.RunWithUi(ctx, comm, ui)
		if cmd.ExitStatus() != 0 {
			return errors.New("error uploading virtual machine; retrying...")
		}
                return err
	})


```

Reply:
AH, that would be much better. Lemme take a look into this.

Reply:
Using this methods in latest commits.